libs/datamodel/core/tests/attributes/postgres_indices/cockroachdb.rs

```diff
@@ -0,0 +1,216 @@
+use crate::{common::*, with_header, Provider};
+
+#[test]
+fn array_field_default_ops() {
+    let dml = indoc! {r#"
+        model A {
+          id Int   @id
+          a  Int[]
+
+          @@index([a], type: Gin)
+        }
+    "#};
+
+    let schema = with_header(dml, Provider::Cockroach, &["extendedIndexes", "cockroachDb"]);
+    let schema = parse(&schema);
+
+    let field = IndexField::new_in_model("a");
+
+    schema.assert_has_model("A").assert_has_index(IndexDefinition {
+        name: None,
+        db_name: Some("A_a_idx".to_string()),
+        fields: vec![field],
+        tpe: IndexType::Normal,
+        defined_on_field: false,
+        algorithm: Some(IndexAlgorithm::Gin),
+        clustered: None,
+    });
+}
+
+#[test]
+fn no_ops_json_prisma_type() {
+    let dml = indoc! {r#"
+        model A {
+          id Int  @id
+          a  Json
+
+          @@index([a], type: Gin)
+        }
+    "#};
+
+    let schema = with_header(dml, Provider::Cockroach, &["extendedIndexes", "cockroachDb"]);
+    let schema = parse(&schema);
+
+    let field = IndexField::new_in_model("a");
+
+    schema.assert_has_model("A").assert_has_index(IndexDefinition {
+        name: None,
+        db_name: Some("A_a_idx".to_string()),
+        fields: vec![field],
+        tpe: IndexType::Normal,
+        defined_on_field: false,
+        algorithm: Some(IndexAlgorithm::Gin),
+        clustered: None,
+    });
+}
+
+#[test]
+fn with_raw_unsupported() {
+    let dml = indoc! {r#"
+        model A {
+          id Int                     @id
+          a  Unsupported("geometry")
+
+          @@index([a], type: Gin)
+        }
+    "#};
+
+    let schema = with_header(dml, Provider::Cockroach, &["extendedIndexes", "cockroachDb"]);
+    let schema = parse(&schema);
+
+    let field = IndexField::new_in_model("a");
+
+    schema.assert_has_model("A").assert_has_index(IndexDefinition {
+        name: None,
+        db_name: Some("A_a_idx".to_string()),
+        fields: vec![field],
+        tpe: IndexType::Normal,
+        defined_on_field: false,
+        algorithm: Some(IndexAlgorithm::Gin),
+        clustered: None,
+    });
+}
+
+#[test]
+fn jsonb_column_as_the_last_in_index() {
+    let dml = indoc! {r#"
+        model A {
+          id Int  @id
+          a  Json
+          b  Int[]
+
+          @@index([b, a], type: Gin)
+        }
+    "#};
+
+    let schema = with_header(dml, Provider::Cockroach, &["extendedIndexes", "cockroachDb"]);
+    let schema = parse(&schema);
+
+    let b = IndexField::new_in_model("b");
+    let a = IndexField::new_in_model("a");
+
+    schema.assert_has_model("A").assert_has_index(IndexDefinition {
+        name: None,
+        db_name: Some("A_b_a_idx".to_string()),
+        fields: vec![b, a],
+        tpe: IndexType::Normal,
+        defined_on_field: false,
+        algorithm: Some(IndexAlgorithm::Gin),
+        clustered: None,
+    });
+}
```

Comment:
Nothing actionable for this PR, but I'm wondering whether the tests with dml assertions are bringing much to the table — it might be enough to assert that the schema is valid.

Reply:
Fully agree.